core-backend/common/src/funcs.rs

```diff
@@ -297,7 +309,12 @@ where
     pub fn reply(ctx: &mut R, payload_ptr: u32, len: u32, value_ptr: u32) -> Result<(), R::Error> {
         let read_payload = ctx.register_read(payload_ptr, len);
         let value = Self::register_and_read_value(ctx, value_ptr)?;
-        let payload = ctx.read(read_payload)?.try_into()?;
+        let payload = ctx
+            .read(read_payload)?
+            .try_into()
+            .map_err(|PayloadSizeError| {
+                TrapExplanation::FallibleExt(MessageError::MaxMessageSizeExceed.into())
+            })?;
```

Comment:
Can you please introduce this repeated stuff in ctx fn maybe?

Reply:
Seems very ad-hoc-ish to me